### ISSUE TITLE
Error creating GL context; Couldn't find any pixel format that matches the criterias.

### ISSUE THREAD

**Issue by chandru2012** [jwilm/alacritty#921]
Which operating system does the issue occur on?
CentOS Linux release 7.3.1611 (Core)
If on linux, are you using X11 or Wayland?
X11

Compiled using : cargo build and run using RUST_BACKTRACE=1 cargo run, following is output:
`[root@DEV_test_master alacritty]# RUST_BACKTRACE=1 cargo run
    Finished dev [unoptimized + debuginfo] target(s) in 0.0 secs
     Running `target/debug/alacritty`
libGL error: unable to load driver: swrast_dri.so
libGL error: failed to load driver: swrast
Alacritty encountered an unrecoverable error:

        Error creating GL context; Couldn't find any pixel format that matches the criterias.

`

**Comment by chandru2012** [comment]
Compiled on centos virtual machine

**Comment by chetgurevitch** [comment]
I get the same issue with mesa 17.3 from the arch testing repos. Works fine with mesa 17.2 though.

**Comment by mebur** [comment]
I get a similar error on Solus.

OpenGL vendor string: Intel Open Source Technology Center
OpenGL renderer string: Mesa DRI Intel(R) HD Graphics 5500 (Broadwell GT2) 
OpenGL core profile version string: 4.5 (Core Profile) Mesa 17.3.1

Error message from "env RUST_BACKTRACE=1 cargo run":
...
Running `target/debug/alacritty`
Alacritty encountered an unrecoverable error:

	Error creating GL context; Couldn't find any pixel format that matches the criterias.

**Comment by chrisduerr** [comment]
If the transparency is the issue, I would assume that there is no support for the rgba pixel format but only the rgb one. So I'd assume that for now, disabling transparency is the only option you have.

**Comment by divoxx** [comment]
I'm seeing the same error message when trying to run alacritty through `optirun` or `primusrun` on a laptop with 2 graphics card (intel/nvidia) using bbswitch/bumblebee.

Alacritty runs fine on the on-board intel but fail with that error when running on the nvidia board. Other software runs fine through nvidia using optirun, so it's not a problem with the setup.

**Comment by tilda** [comment]
I updated my mesa to oibaf's ppa (along with all the other video stuff) and have this now
```
 ~ alacritty
Alacritty encountered an unrecoverable error:

	Error creating GL context; Couldn't find any pixel format that matches the criterias.
```

**Comment by lo48576** [comment]
I have the same error on my machine, but by this patch (<https://github.com/lo48576/alacritty/commit/b274a9f5055115b152159c0f146542a91fb4bb63>), it works as expected with transparency enabled.
I'm not sure whether this is the really same problem as yours, but you can try this.

* kernel: linux-4.15.9 (gentoo)
* mesa: mesa-18.0.0_rc4
* glxinfo (snip):
  ```
  OpenGL vendor string: Intel Open Source Technology Center
  OpenGL renderer string: Mesa DRI Intel(R) Haswell Mobile
  OpenGL core profile version string: 4.5 (Core Profile) Mesa 18.0.0-rc4
  ```

**Comment by mebur** [comment]
The patch above solves the issue for me! Thanks!

**Comment by chrisduerr** [comment]
@lo48576 Testing it on my machine which doesn't have that issue, everything still works.

So if this is just an improvement without disadvantage, maybe it should be upstreamed? Unless you know about an issue it would probably be nice to have a PR for this.

**Comment by tilda** [comment]
I can confirm, indeed, that the patch works flawlessly.

**Comment by lluixhi** [comment]
@lo48576 
Can confirm, that patch is enough to get alacritty working on gentoo for me, with mesa-18.0.0_rc4

**Comment by tbehner** [comment]
Funnily, the patch results in the exact same crash it's supposed to fix. 
If I switch with_srgb(false), everything works as usual....

**Comment by alexherbo2** [comment]
I have the same issue since #1178 has been merged and using [Mesa 17.3.6][].

[Mesa 17.3.6]: https://github.com/mesa3d/mesa/releases/tag/mesa-17.3.6

**Comment by lo48576** [comment]
@tbehner @alexherbo2 I created Pull Request #1178 just now, this should solve your problem.
